### PR TITLE
feat(HACBS-2381): update service to use operator-toolkit

### PR DESCRIPTION
**Pull request by davidmogar**
The operator-toolkit has been already released and operator-goodies is deprecated.

### CODE REVIEW

File: api/v1alpha1/release_webhook.go [hunk 1]
```diff
@@ -17,38 +17,44 @@ limitations under the License.
...
+	if !reflect.DeepEqual(newRelease.Spec, oldRelease.Spec) {
 		return fmt.Errorf("release resources spec cannot be updated")
 	}
 
 	return nil
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
-func (r *Release) ValidateDelete() error {
+func (w *ReleaseWebhook) ValidateDelete(ctx context.Context, obj runtime.Object) error {
```
Comment (johnbieren) [inline api/v1alpha1/release_webhook.go:58]: Why do we need to add these arguments?

File: api/v1alpha1/release_webhook.go [hunk 2]
```diff
@@ -17,38 +17,44 @@ limitations under the License.
...
 		Complete()
 }
 
 //+kubebuilder:webhook:path=/validate-appstudio-redhat-com-v1alpha1-release,mutating=false,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases,verbs=create;update,versions=v1alpha1,name=vrelease.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Validator = &Release{}
-
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (r *Release) ValidateCreate() error {
+func (w *ReleaseWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
```
Comment (johnbieren) [inline api/v1alpha1/release_webhook.go:41]: Why do we need to add these arguments?

File: api/v1alpha1/releaseplan_webhook.go [hunk 3]
```diff
@@ -17,57 +17,64 @@ limitations under the License.
...
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
-func (rp *ReleasePlan) ValidateUpdate(old runtime.Object) error {
-	return rp.validateAutoReleaseLabel()
+func (w *ReleasePlanWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) error {
+	return w.validateAutoReleaseLabel(newObj)
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (rp *ReleasePlan) ValidateDelete() error {
+func (w *ReleasePlanWebhook) ValidateDelete(ctx context.Context, obj runtime.Object) error {
```
Comment (johnbieren) [inline api/v1alpha1/releaseplan_webhook.go:70]: same question about the new parameters

File: api/v1alpha1/webhook_suite_test.go [hunk 4]
```diff
@@ -50,7 +51,7 @@ var mgr manager.Manager
 
 func TestAPIs(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "Release Webhook Suite")
+	RunSpecs(t, "Release ReleasePlanWebhook Suite")
```
Comment (johnbieren) [inline api/v1alpha1/webhook_suite_test.go:54]: This doesn't seem right

File: loader/loader_mock_test.go [hunk 5]
```diff
@@ -21,67 +20,10 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		loader = NewMockLoader()
 	})
 
-	When("calling getMockedResourceAndErrorFromContext", func() {
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply loader/loader_mock_test.go:24]: Yes.

File: api/v1alpha1/release_webhook.go (hunk 2 above)
Comment (davidmogar) [reply api/v1alpha1/release_webhook.go:41]: Because it's an interface.

**Review by johnbieren — APPROVED** [review]

**Comment by johnbieren** [comment]
CI isn't going to pass until https://github.com/redhat-appstudio/e2e-tests/pull/634 is merged

**Comment by davidmogar** [comment]
> CI isn't going to pass until [redhat-appstudio/e2e-tests#634](https://github.com/redhat-appstudio/e2e-tests/pull/634) is merged

Thanks for the heads up. I forgot about your change being merged.

**Review by kasemAlem — APPROVED** [review]

**Pull request merged**